force-app/main/default/classes/StaticTask.cls

```diff
@@ -9,30 +9,103 @@ public with sharing class StaticTask {
   . Then it parses data and return Map collection where key is meail and value is Contact
   Feel free to refactorthis class. Don't forget about single responsibility approach when you create a new method.
   */
+  
   public static Map<String, Contact> extractEmailToContactMap(String payload) {
+
     Map<String, Contact> emailToContactMap = new Map<String, Contact>();
-    List<String> contacts = payload.split(';');
+
+    List<String> contacts = getContactsAsStringList(payload);
+
+    setContactsToMap(contacts, emailToContactMap);
+
+    return emailToContactMap;
+  }
+
+  public static void setContactsToMap(List<String> contacts, Map<String, Contact> emailToContactMap){
     for (String contactAsString : contacts) {
+      
       Contact parsedContact = new Contact();
-      List<String> contactSeparatedByComaString = contactAsString.split(',');
-      for (String element : contactSeparatedByComaString) {
-        if (element.contains('FirstName')) {
-          parsedContact.FirstName = element.split(':')[1].replaceAll(' ', '');
-        }
-        if (element.contains('LastName')) {
-          parsedContact.LastName = element.split(':')[1].replaceAll(' ', '');
-        }
-        if (element.contains('Email')) {
-          if (String.isNotBlank(element.split(':')[1])) {
-            parsedContact.Email = element.split(':')[1].replaceAll(' ', '');
-            emailToContactMap.put(
-              element.split(':')[1].replaceAll(' ', ''),
-              parsedContact
-            );
-          }
-        }
+      
+      List<String> contactSeparatedByComaString = getContactSeparatedByComaString(contactAsString);
+      
+      setContactToMap(contactSeparatedByComaString, parsedContact, emailToContactMap);
+    }
+  }
+
+  public static void setContactToMap(List<String> contactSeparatedByComaString, Contact parsedContact, Map<String, Contact> emailToContactMap){
```

Comment:
it looks like you can simplify a little bit logic by moving Map<String, Contact> emailToContactMap to static variable to avoid passing same value in each method